mlflow/utils/autologging_utils/__init__.py

```diff
@@ -547,13 +547,33 @@ def should_log(self):
                class.
             """
             return (self._parent is None) or (
-                self._parent.allow_children and self._parent.clazz != self.clazz
+                self._parent.allow_children and self._parent.estimator is not self.estimator
             )
 
+        def should_log(self):
```

Comment:
When should we use `should_log` instead of `should_enable_patch`?

Reply:
e.g., inside `lightgbm.train`, when we calls `LGBMClassifierInstance.fit`, the calling stack is like
```
(1) patched LGBMClassifier.fit ->  (should_enable_patch=True)
(2) original LGBMClassifier.fit -> 
(3) patched LGBMModel.fit ->  (should_enable_patch=False)
(4) original LGBMModel.fit ->
(5) patched lightgbm.train ->  (should_log = True)
(5) original lightgbm.train
```

Reply:
@WeichenXu123 can `should_log` handle all of this? Still unclear why we need 2 separate methods.

Reply:
we can.

Reply:
Awesome! LGTM once we do that :)

Reply:
Adding an argument like:
`def should_log(check_first_session_of_current_estimator=False)` ?

Reply:
Still a bit confused about why we need this argument. Is the following implementation sufficient?

```
def should_log():
      for training_session in TrainingSession._session_stack:
          if training_session == self:
              break
          elif training_session.estimator == self.estimator:
              return training_session.should_log()

       return self._parent is None or self._parent.allow_children
```

Reply:
Yes. I adopt your `should_log` code.
But, it introduce a minor issue.